src/test/java/com/puppycrawl/tools/checkstyle/bdd/InlineConfigParser.java

```diff
@@ -50,9 +50,22 @@ public final class InlineConfigParser {
     /** A pattern matching the symbol: "\" or "/". */
     private static final Pattern SLASH_PATTERN = Pattern.compile("[\\\\/]");
 
-    /** A pattern to find the string: "// violation". */
+    /**
+     * A pattern that matches the following comments formats.
+     * <ol>
+     *     <li> "// violation" </li>
+     *     <li> "// violation, 'violation message'" </li>
+     *     <li> "// violation 'violation messages'" </li>
+     * </ol>
+     *
+     * <p>
+     * This pattern will not match comments of the form "// violation, explanation" or
+     * "// violation, explanation, 'violation message'". These are matched by
```

Comment:
a bit more visual clarity:
```
This pattern will not match comments of the form 
"// violation, explanation" or
"// violation, explanation, 'violation message'".
These are matched by
```

Reply:
Done, just made another list since newlines will not translate to javadoc, ordered lists look nice in IDE hints:
![image](https://github.com/checkstyle/checkstyle/assets/31252532/65fe0084-9613-4b8d-b98b-7526c2697cb3)